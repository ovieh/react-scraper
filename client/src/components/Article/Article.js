import React, { useState } from 'react';
import { ListGroupItem, 
  ListGroupItemHeading, 
  ListGroupItemText, 
  Button,
  Collapse, Container, Row, Col } from 'reactstrap';
import Comment from '../Comment';


const Article = ( {headline, summary, url, index, buttonText, handleSubmit, id, article, unsave, image, comments} ) => {
  const [collapse, setCollapse] = useState(false);
  const toggle = () => {
    setCollapse(!collapse);
  }

  const buttonStyle= {
    marginBottom: 30
  }


  return (
    <ListGroupItem key={index}>
      {
        unsave &&
        <Button onClick={() => unsave(article)}  close />
      }
      <Container>
        <Row>
          <Col lg='3'md='4' s="6">
            <img src={image} alt="" className="image-fluid "/>
          </Col>
          <Col lg='9' md='8's="6">
            <ListGroupItemHeading>{headline}</ListGroupItemHeading>
            <ListGroupItemText>{summary}</ListGroupItemText>
            
            <Container>
              <Row>
                <Col xs='auto'>
                  <Button href={`https://www.nytimes.com${url}`} rel="noopener noreferrer" target="_blank" className="mb-1">View on NYTimes</Button>
                </Col>
                {" "}
                <Col xs='auto'>
                  {
                    unsave ? <Button onClick={() => toggle()} >{buttonText}</Button>
                          : <Button onClick={() => handleSubmit(article)}>{buttonText}</Button>
                            
                  }
                </Col>
              </Row>
            </Container>
           
          </Col>
        </Row>
      </Container>

      <Collapse isOpen={collapse}>
        <Comment id={id} />
      </Collapse>
    </ListGroupItem>
  );

};

export default Article;