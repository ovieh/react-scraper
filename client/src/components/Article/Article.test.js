import React from 'react';
import Article from './Article';
import { create } from 'react-test-renderer';

describe("Article component", () => {
  test('it matches snapshot', () => {
    const component = create(
    <Article text="render me" />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});

