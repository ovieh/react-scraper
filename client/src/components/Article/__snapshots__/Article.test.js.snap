// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component it matches snapshot 1`] = `
<li
  className="list-group-item"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-3"
      >
        <img
          alt=""
        />
      </div>
      <div
        className="col-9"
      >
        <h5
          className="list-group-item-heading"
        />
        <p
          className="list-group-item-text"
        />
        <a
          aria-label={null}
          className="btn btn-secondary"
          href="https://www.nytimes.comundefined"
          onClick={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          View on NYTimes
        </a>
         
        <button
          aria-label={null}
          className="btn btn-secondary"
          onClick={[Function]}
          type="button"
        />
      </div>
    </div>
  </div>
  <div
    className="collapse"
    style={Object {}}
  >
    <div
      className="card"
    >
      <div
        className="card-body"
      >
        <form
          className=""
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label
              className=""
              htmlFor="comments"
            >
              Comments
            </label>
            <input
              className="form-control"
              id="comments"
              name="comment"
              onChange={[Function]}
              type="text"
              value=""
            />
            <button
              aria-label={null}
              className="btn btn-secondary"
              onClick={[Function]}
              style={
                Object {
                  "marginTop": "2px",
                }
              }
              type="submit"
              value="submit"
            >
              Submit
            </button>
          </div>
          <ul
            className="list-group"
          />
        </form>
      </div>
    </div>
  </div>
</li>
`;
